cibuildwheel/windows.py

```diff
@@ -115,6 +122,50 @@ def install_pypy(tmp: Path, arch: str, url: str) -> Path:
     return installation_path / "python.exe"
 
 
+def setup_setuptools_cross_compile(
+    python_configuration: PythonConfiguration,
+    python_libs_base: Path,
+    env: Dict[str, str],
+) -> None:
+    # We write to distutils_cfg for distutils-based builds because we know we
+    # currently don't have one (unless it's our own from a previous build)
+    for p in env["PATH"].split(os.pathsep):
+        distutils_cfg = Path(p) / "Lib/site-packages/setuptools/_distutils/distutils.cfg"
+        if distutils_cfg.parent.is_dir():
+            break
+    else:
+        log.warning("Did not find setuptools install to configure")
+        return
```

Comment:
The problem with this new logic is that `call(["python" ...` doesn't actually search `PATH` to resolve `python`. It first looks in the current executable directory (normal Windows behaviour, assuming that you want the files you installed with your own app before searching other installs). So it only works if we're already running in the correct Python, in which case we may as well go straight to `sys.prefix`.

Also, if we're going to run Python and import it, I'd rather `import distutils; print(distutils.__file__)` and work from that to save a few assumptions. Or deliberately keep the venv path around from earlier - I tried that briefly and decided it wasn't worth the changes, but it's going to be the most reliable.

Reply:
Oh, and we alse know that `env["PATH"]` contains the right venv first because we just updated it ourselves.

Reply:
> The problem with this new logic is that `call(["python" ...` doesn't actually search `PATH` to resolve `python`. It first looks in the current executable directory (normal Windows behaviour, assuming that you want the files you installed with your own app before searching other installs).

🤯 whoa really? when you say the 'current executable directory', do you mean the working directory, which would be the project dir in this case? We do `call(["python" ...` all over the place and we've never had an issue with the wrong one being invoked. Maybe because people don't tend to have a `python.exe` in the root of their project.

> Also, if we're going to run Python and import it, I'd rather `import distutils; print(distutils.__file__)` and work from that to save a few assumptions.

I'd prefer that, I think. We probably won't preinstall setuptools in the build venv forever. Though I think it should be `import setuptools._distutils; print(setuptools._distutils.__file__)`, right? (though, depending on the outcome of the isolated build-backend venv, it might be a moot point...)